apps/omg_watcher/lib/omg_watcher/block_getter.ex

```diff
@@ -98,15 +99,16 @@ defmodule OMG.Watcher.BlockGetter do
     {:reply, Core.chain_ok(state), state}
   end
 
-  def handle_cast(
-        {:apply_block,
+  def handle_cast({:apply_block, %BlockApplication{} = to_apply}, state), do: apply_block(to_apply, state)
+
+  defp apply_block(
```

Comment:
solving `apply_block`'s metrics is the issue I mentioned that needs to be figured out.

Somewhere probably here, you need to define a `background`-namespace transaction that will encompass processing of a single block - so sounds like

```
@decorate transaction 
def apply_block
```

or
```
@decorate transaction 
defp do_apply_block
```